### PR TITLE
Update `TrainConfig` optimizer hyperparameters

### PR DESCRIPTION
**Pull request by jon-tow**
This PR provides the following updates based on observations from issue #53:
- Adds the previously unused `weight_decay` TrainConfig parameter to `AccelerateRLModel` optimizer initialization.
- Adds an `opt_eps` TrainConfig parameter to the `AccelerateRLModel` to allow users to specify an optimizer's epsilon value.
- Renames `learning_rate_init` and `learning_rate_target` to `lr_init` and `lr_target` TrainConfigs, respectively, for consistency with learning rate naming conventions in `torch.optim` algorithms.
- Removes unused `lr_ramp_steps` and `lr_decay_steps` TrainConfig parameters which look to be leftovers from the [magicCARP lr scheduler](https://github.com/EleutherAI/magiCARP/blob/777a578abcf0f790b653d3bce9dcbff8f5cd6bd7/carp/util.py#L25).

### CODE REVIEW

**Comment by LouisCastricato** [comment]
@herbiebradley since this is for your issue.

**Review by herbiebradley — APPROVED** [review]
Looks good! Can we change the default epsilon to `1.0e-7` (default in Tensorflow) since the ablation paper found it was significantly better?

Also, does anyone know where the weight decay of `1.0e-6` comes from, since it is significantly lower than the default?

**Comment by reciprocated** [comment]
> Also, does anyone know where the weight decay of `1.0e-6` comes from, since it is significantly lower than the default?

From here https://github.com/EleutherAI/magiCARP/blob/3537b4d7c98a4a384964856740a65731746e04fe/configs/base_config.yml#L18

**Review by reciprocated — COMMENTED** [review]
lgtm!

**Review by LouisCastricato — APPROVED** [review]

**Comment by jon-tow** [comment]
@herbiebradley I intentionally left the default $\epsilon$ to `1.0e-8` as I wasn't 100% sure that the improvement would directly carry over to AdamW (we do not use vanilla Adam). I'd like to try some experiments before updating in another PR if that's okay.

**Comment by jon-tow** [comment]
cc #76 for awareness

**Comment by ayulockin** [comment]
Thanks for tagging the issue @jon-tow.
